trace-utils/src/tracer_payload.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::cmp::Ordering;
+
+use crate::trace_utils::cmp_send_data_payloads;
+use datadog_trace_protobuf::pb::{Span, TracerPayload};
+
+type TracerPayloadV04 = Vec<Span>;
+
+#[derive(Debug, Clone)]
+/// Enumerates the different encoding types.
+/// ```
+pub enum TraceEncoding {
+    /// v0.4 endoding (TracerPayloadV04).
+    V04,
+    /// v0.7 endoding (TracerPayload).
+    V07,
+}
+
+#[derive(Debug, Clone)]
+/// Enum representing a general abstraction for a collection of tracer payloads.
+/// ```
+pub enum TracerPayloadCollection {
```

Comment:
I've used an enum to encapsulate the abstraction for using both encoding types in SendData. I thought about using generics and traits but this way seemed less convoluted. Let me know what you think.

Reply:
I think this is a straight forward way to represent it. We can always revisit it if we feel the need for generics.